### PR TITLE
state: Store and load system repo groups metadata

### PR DESCRIPTION
**Pull request by m-blaha**
Comps metadata for each installed group is now stored in the system state and loaded into libsolv pool from there.

### CODE REVIEW

File: libdnf/repo/solv_repo.cpp [hunk 1]
```diff
@@ -277,6 +278,49 @@ void SolvRepo::load_repo_main(const std::string & repomd_fn, const std::string &
...
+            auto & system_state = base->p_impl->get_system_state();
+            auto comps_dir = system_state.get_group_xml_dir();
+            auto installed_groups = system_state.installed_groups();
+            for (const auto & group_id : installed_groups) {
+                auto ext_fn = comps_dir / (group_id + ".xml");
+                fs::File ext_file;
+                try {
+                    ext_file = fs::File(ext_fn, "r", true);
+                } catch (std::filesystem::filesystem_error & e) {
+                    logger.debug(
```
Comment (jrohel) [inline libdnf/repo/solv_repo.cpp:300]: Is this really a "debug" message? Is it logging something that is standard behavior?  It looks more like a warning. We want to warn the user that a loading error has occurred.

File: libdnf/system/state.cpp [hunk 2]
```diff
@@ -315,6 +315,15 @@ std::set<std::string> State::get_package_groups(const std::string & name) {
     }
 }
 
+std::vector<std::string> State::installed_groups() {
```
Comment (jan-kolarik) [inline libdnf/system/state.cpp:318]: I would prefer ``get_installed_groups`` here.

**Review by jan-kolarik — CHANGES_REQUESTED** [review]
I left some comments above, they are addressing some minor issues from my point of view. Also a rebase is needed now. Otherwise LGTM.

File: libdnf/base/transaction.cpp [hunk 3]
```diff
@@ -616,6 +618,8 @@ Transaction::TransactionRunResult Transaction::Impl::run(
                     }
                 }
                 system_state.set_group_state(group.get_groupid(), state);
+                // save the current xml group definition
+                group.dump(comps_xml_dir / (group.get_groupid() + ".xml"));
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply libdnf/base/transaction.cpp]: I remember that python json and xml libraries use `dump()` method name for printing the content of element to string / file. But I like your suggestion and I've added a commit for renaming the methods.

File: libdnf/system/state.cpp (hunk 2 above)
Comment (m-blaha) [reply libdnf/system/state.cpp]: Renamed.

**Pull request merged**